tests/integration/test_s3.py

```diff
@@ -60,25 +60,48 @@ def get_method(self, *args, **kwargs):
         return 'PUT'
 
 
+# def test_host_and_path_addressing(wrapped):
+#     """ Decorator that runs a test method with both - path and host style addressing. """
+#     # TODO - needs to be fixed below!
+#     def wrapper(self):
+#         try:
+#             # test via path based addressing
+#             TestS3.OVERWRITTEN_CLIENT = aws_stack.connect_to_service('s3', config={'addressing_style': 'virtual'})
+#             wrapped()
+#             # test via host based addressing
+#             TestS3.OVERWRITTEN_CLIENT = aws_stack.connect_to_service('s3', config={'addressing_style': 'path'})
+#             wrapped()
+#         finally:
+#             # reset client
+#             TestS3.OVERWRITTEN_CLIENT = None
+#     return
+
 class TestS3(unittest.TestCase):
+    OVERWRITTEN_CLIENT = None
+
     def setUp(self):
-        self.s3_client = aws_stack.connect_to_service('s3')
+        self._s3_client = aws_stack.connect_to_service('s3')
         self.sqs_client = aws_stack.connect_to_service('sqs')
 
+    @property
+    def s3_client(self):
+        return TestS3.OVERWRITTEN_CLIENT or self._s3_client
+
     def test_create_bucket_via_host_name(self):
         body = """<?xml version="1.0" encoding="UTF-8"?>
             <CreateBucketConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
                 <LocationConstraint>eu-central-1</LocationConstraint>
             </CreateBucketConfiguration>"""
         headers = aws_stack.mock_aws_request_headers('s3')
         bucket_name = 'test-%s' % short_uid()
-        headers['Host'] = '%s.s3.amazonaws.com' % bucket_name
+        headers['Host'] = '%s.s3.localhost.localstack.cloud:4566' % bucket_name
```

Comment:
Let's better use `S3_VIRTUAL_HOSTNAME` here, to avoid hardcoding values:
```
        headers['Host'] = '%s.%s:%s' % (bucket_name, S3_VIRTUAL_HOSTNAME, config.EDGE_PORT)
```
(Perhaps this could be moved to a small util function like `get_bucket_hostname(bucket_name)`..?)